### PR TITLE
implemented eigen_values function

### PR DESCRIPTION
**Pull request by ranjithkumar007**

### CODE REVIEW

**Comment by ranjithkumar007** [comment]
[Error](https://travis-ci.org/symengine/symengine/jobs/261413770#L9176) message - `cat: write error: No space left on device`

**Comment by srajangarg** [comment]
yes, we can try that. push a commit here to see if it works.

**Comment by srajangarg** [comment]
@isuruf what's eating up all the disk space? Afaik, travis gives us 20GB per job.

**Comment by isuruf** [comment]
Coverage info

File: bin/install_travis.sh [hunk 1]
```diff
@@ -110,7 +110,7 @@ source activate $our_install_dir;
 
 if [[ "${WITH_FLINT_DEV}" == "yes" ]] && [[ "${WITH_ARB}" != "yes" ]]; then
     git clone https://github.com/wbhart/flint2;
-    cd flint2 && git checkout 06defcbc52efe41a8c73496ffde9fc66941e3f0d && ./configure --prefix=$our_install_dir --with-gmp=$our_install_dir --with-mpfr=$our_install_dir && make -j8 install && cd ..;
+    cd flint2 && git checkout 06defcbc52efe41a8c73496ffde9fc66941e3f0d && ./configure --prefix=$our_install_dir --with-gmp=$our_install_dir --with-mpfr=$our_install_dir && make -j8 install > /dev/null && cd ..;
```
Comment (certik) [inline bin/install_travis.sh]: That's fine for now. But ultimately we should just put this into Conda and use Conda.

**Comment by certik** [comment]
Now the test fails with:
```
+++conda config --set always_yes yes --set changeps1 no

+++'[' -x /usr/lib/command-not-found ']'

+++/usr/lib/command-not-found -- conda

conda: command not found
```
So something broke on Travis.

File: bin/install_travis.sh (hunk 1 above)
Comment (isuruf) [reply bin/install_travis.sh]: Why this change? I don't see the need for this commit.

**Comment by ranjithkumar007** [comment]
@certik , this PR has the same problem that #1305 had. I got it fixed by moving piranha to a different build. The problem was with coverage info as pointed out by @isuruf.  Should I wait for #1305 to get merged in and then update this PR or add the commits that fixed the error here?

**Comment by certik** [comment]
@ranjithkumar007 does not matter to me, but #1305 can't be merged without more work it seems. So just concentrate on one PR, and let's get it merged, then the other one.

**Comment by ranjithkumar007** [comment]
Thanks @certik , I updated the PR.
Travis should pass now.

**Comment by ranjithkumar007** [comment]
@certik , Travis Passed :smiley:

**Comment by certik** [comment]
@ranjithkumar007 thanks!

**Comment by ranjithkumar007** [comment]
Thanks a lot @certik. :smiley:

**Pull request merged**